codec-http/src/main/java/io/netty/handler/codec/http/HttpResponseStatus.java

```diff
@@ -447,7 +449,21 @@ public static HttpResponseStatus valueOf(int code) {
     }
 
     /**
-     * Parses the specified HTTP status line into a {@link HttpResponseStatus}.  The expected formats of the line are:
+     * Returns the {@link HttpResponseStatus} represented by the specified {@code code} and {@code reasonPhrase}.
+     * If the specified code is a standard HTTP status {@code code} and {@code reasonPhrase}, a cached instance
+     * will be returned. Otherwise, a new instance will be returned.
+     * @param code The response code value.
+     * @param reasonPhrase The response code reason phrase.
+     * @return the {@link HttpResponseStatus} represented by the specified {@code code} and {@code reasonPhrase}.
+     */
+    public static HttpResponseStatus valueOf(int code, String reasonPhrase) {
+        HttpResponseStatus responseStatus = valueOf(code);
+        return responseStatus.reasonPhrase().contentEquals(reasonPhrase) ? responseStatus :
+                new HttpResponseStatus(code, reasonPhrase);
```

Comment:
@Scottmitch If the `code` is not represented in this class, will happen a double instatuation of `HttpResponseStatus`: first in `valueOf(code)`, and second at this line. I think we can avoid this. WDYT?

Reply:
good catch ... yes we can avoid the double allocation